aspnet/fundamentals/distributed-cache.rst

```diff
@@ -1,10 +1,106 @@
+
+Working with a Redis instance of ``IDistributedCache`` should be completely transparent to your application code. You should request an instance of ``IDistributedCache``, as shown above, and use that interface to set and get values from the cache. The only place in your application where you should reference the Redis implementation is in your ``ConfigureServices`` method, where you will configure the Redis implementation.
+
+In the sample code associated with this article, a ``RedisCache`` implementation is used when the server is configured for a ``Staging`` environment. Thus ``ConfigureStagingServices`` has the required code to configure the ``RedisCache``:
+
+.. literalinclude:: dist-cache/sample/src/DistCacheSample/Startup.cs
+  :language: c#
+  :linenos:
+  :lines: 27-40
+  :emphasize-lines: 8-13
+
+.. note:: To install Redis on your local machine, install the chocolatey package http://chocolatey.org/packages/redis-64/ and run ``redis-server`` from a command prompt.
+
+Using a SQL Server Distributed Cache
+------------------------------------
+The `SqlServerCache <https://github.com/aspnet/Caching/tree/1.0.0-rc1/src/Microsoft.Extensions.Caching.SqlServer>`_ implementation allows the distributed cache to use a SQL Server database as its backing store. The installation script installs a table with whatever name you configure and the following schema:
+
+.. image:: dist-cache/_static/SqlServerCacheTable.png
+
+.. note:: If you're working with the RC1 version of ``SqlServerCache``, there isn't a working installer. You can install the required table using the scripts found in the `SqlQueries.cs file <https://github.com/aspnet/Caching/blob/1.0.0-rc1/src/Microsoft.Extensions.Caching.SqlConfig/SqlQueries.cs>`_. This will be addressed in RC2.
+
+Just as with a Redis implementation, working with a ``SqlServerCache`` implementation should be transparent to your application code. Your application code should simply request an instance of type ``IDistributedCache`` and get and set cached values through this interface. You should only configure ``SqlServerCache`` in your application's ``ConfigureServices`` method.
+
+In the sample code associated with this article, a ``SqlServerCache`` implementation is used when the server is configured for a ``Production`` environment. Thus, ``ConfigureProductionServices`` has the required code to configure the ``SqlServerCache``:
+
+.. literalinclude:: dist-cache/sample/src/DistCacheSample/Startup.cs
+  :language: c#
+  :linenos:
+  :lines: 42-56
+  :emphasize-lines: 9-14
+
+.. note:: The ``ConnectionString`` (and optionally, ``SchemaName`` and ``TableName``) should typically be stored in :doc:`configuration` in real-world applications, rather than hard-coded here where a code change is required to update the values.
+
+Recommendations
+---------------
```

Comment:
Avoid using the in-memory implementation (``MemoryCache``) in multi-machine environments.